Add 'time-out' message when a command timed out

veger opened the pull request:
This PR clarifies that a time-out occurred, solving confusion by the end user. See #630

zmb3 commented:
I think the executor should return stderr exactly as the process spit out, and we should detect the timeout downstream and display an appropriate message.

Are you up for making these changes?

veger commented:
Note that only the implements command checks (downstream) if the failing reason is a time-out. Other commands (check 'usage' only) does not (seem to) check this.

Should we update them all?

zmb3 commented:
gorename and tester also check for timeouts.  You could also add a check to usage.

zmb3 commented on lib/implements/implements.js:
```diff
@@ -55,8 +55,11 @@ export default class Implements {
-            if (this.view) {
-              this.view.update('guru failed' + os.EOL + os.EOL + r.stderr.trim())
+              if (r.exitcode === 124) {
+                this.view.update('guru failed: time-out')
```
How about:

```js
guru failed: operation timed out after `${operation.timeout} ms`
```

veger commented:
Updated 'implements' time-out with nicer message (and fixed position of return statement, found by linter :smile: ) and added similar implementation to 'usage'

zmb3 commented:
Thanks!

tpjg commented:
Can this be added for more commands? E.g. when go-plus is doing it's lintTest ("go test -c -o {tempdir} .") it may also timeout, especially if CGO is involved.
